scripts/swupdate/merge.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/sh
+
+#Below function will return value corresponding to requested key
+function getProperty {
+   PROP_KEY=$1
+   PROPERTY_FILE=$2
+   FILE_TYPE=$3
+   if [ "$fileType" = "properties" ]
+   then
+       PROP_VALUE=`cat $PROPERTY_FILE | grep "$PROP_KEY=" | cut -d'=' -f2`
+   else
+       PROP_VALUE=`cat $PROPERTY_FILE | grep "$PROP_KEY:" | cut -d':' -f2`
+   fi
+   IFS=$'#'
+   read -ra prop_val <<< "$PROP_VALUE"
+   echo "${prop_val[0]// /}"
+}
+
+#Below method perform merging of config
+function performMerge {
+while read line; do
+if [ "$fileType" = "properties" ]
+then
+    IFS=$'='
+else
+    IFS=$':'
+fi
+read -ra ID <<< "$line"
+key="${ID[0]// /}" #remove leading and trailing spaces
+value="${ID[1]// /}"
+if ! grep -Fx "${key}" $unsafeAttributesFile >/dev/null
+then
+if [[ $key != \#* ]]; then #Avoiding commented lines or keys
+configValue=$(getProperty "$key" "$configFile" "$fileType")
+oldSampleValue=$(getProperty "$key" "$oldSampleFile" "$fileType")
+newSampleValue=$(getProperty "$key" "$newSampleFile" "$fileType")
+#if config and old sample value same or new parameter added then replace config value with new sample value
```

Comment:
In the code below, not sure how it handles the "new added parameter".  If there is new parameter in newSampleFile, we have to add new line in configFile -- but it is not being done.

Reply:
Handled. please check.

Reply:
Thanks.